### PR TITLE
Unified the commenting form of the internal properties and removed th…

### PR DESCRIPTION
**Pull request by robertsipka**
…e unused types.

JerryScript-DCO-1.0-Signed-off-by: Robert Sipka rsipka.uszeged@partner.samsung.com

### CODE REVIEW

File: jerry-core/ecma/base/ecma-globals.h [hunk 1]
```diff
@@ -132,47 +130,29 @@ typedef enum
...
-  ECMA_INTERNAL_PROPERTY_NON_INSTANTIATED_BUILT_IN_MASK_32_63,
+  ECMA_INTERNAL_PROPERTY_NON_INSTANTIATED_BUILT_IN_MASK_32_63, /**< Bit-mask of non-instantiated
+                                                                * built-in's properties (bits 32-63)
+                                                                */
 
-  /**
-   * RegExp bytecode array
-   */
-  ECMA_INTERNAL_PROPERTY_REGEXP_BYTECODE,
+  ECMA_INTERNAL_PROPERTY_REGEXP_BYTECODE, /**< RegExp bytecode array */
```
Comment (zherczeg) [inline jerry-core/ecma/base/ecma-globals.h]: Move this to ECMA_INTERNAL_PROPERTY_CODE_BYTECODE

File: jerry-core/ecma/base/ecma-globals.h [hunk 2]
```diff
@@ -132,47 +130,29 @@ typedef enum
...
   ECMA_INTERNAL_PROPERTY_BOUND_FUNCTION_BOUND_THIS,
   ECMA_INTERNAL_PROPERTY_BOUND_FUNCTION_BOUND_ARGS,
 
-  /**
-   * Bit-mask of non-instantiated built-in's properties (bits 0-31)
-   */
-  ECMA_INTERNAL_PROPERTY_NON_INSTANTIATED_BUILT_IN_MASK_0_31,
+  ECMA_INTERNAL_PROPERTY_NON_INSTANTIATED_BUILT_IN_MASK_0_31, /**< Bit-mask of non-instantiated
+                                                               * built-in's properties (bits 0-31)
+                                                               */
```
Comment (zherczeg) [inline jerry-core/ecma/base/ecma-globals.h]: Do we still need these two numbers?

File: jerry-core/ecma/base/ecma-globals.h (hunk 1 above)
Comment (robertsipka) [inline jerry-core/ecma/base/ecma-globals.h]: I've changed the order.

File: jerry-core/ecma/base/ecma-globals.h (hunk 2 above)
Comment (robertsipka) [inline jerry-core/ecma/base/ecma-globals.h]: Yes, the try_to_instantiate_property routines uses this properties.

File: jerry-core/ecma/base/ecma-globals.h [hunk 3]
```diff
@@ -120,59 +120,39 @@ typedef uint32_t ecma_value_t;
 typedef enum
 {
   ECMA_INTERNAL_PROPERTY_CLASS, /**< [[Class]] */
-  ECMA_INTERNAL_PROPERTY_PROTOTYPE, /**< [[Prototype]] */
-  ECMA_INTERNAL_PROPERTY_EXTENSIBLE, /**< [[Extensible]] */
   ECMA_INTERNAL_PROPERTY_SCOPE, /**< [[Scope]] */
   ECMA_INTERNAL_PROPERTY_PARAMETERS_MAP, /**< [[ParametersMap]] */
   ECMA_INTERNAL_PROPERTY_CODE_BYTECODE, /**< first part of [[Code]] - compressed pointer to bytecode array */
```
Comment (zherczeg) [inline jerry-core/ecma/base/ecma-globals.h]: Update comment: pointer to compact bytecode array

File: jerry-core/ecma/base/ecma-globals.h [hunk 4]
```diff
@@ -120,59 +120,39 @@ typedef uint32_t ecma_value_t;
...
+  ECMA_INTERNAL_PROPERTY_REGEXP_BYTECODE, /**< RegExp bytecode array */
```
Comment (zherczeg) [inline jerry-core/ecma/base/ecma-globals.h]: Update comment: pointer to RegExp bytecode array

File: jerry-core/ecma/base/ecma-globals.h [hunk 5]
```diff
@@ -120,59 +120,39 @@ typedef uint32_t ecma_value_t;
...
-
-  /** Implementation-defined identifier of built-in routine
-      that corresponds to a built-in function object
-      ([[Built-in routine's description]]) */
-  ECMA_INTERNAL_PROPERTY_BUILT_IN_ROUTINE_DESC,
+  ECMA_INTERNAL_PROPERTY_BUILT_IN_ID, /**< Implementation-defined identifier of built-in object */
 
-  /** Identifier of implementation-defined extension object */
-  ECMA_INTERNAL_PROPERTY_EXTENSION_ID,
+  ECMA_INTERNAL_PROPERTY_BUILT_IN_ROUTINE_DESC,   /**< Implementation-defined identifier of built-in routine
```
Comment (zherczeg) [inline jerry-core/ecma/base/ecma-globals.h]: I would put ECMA_INTERNAL_PROPERTY_BUILT_IN_ROUTINE_DESC before ECMA_INTERNAL_PROPERTY_NON_INSTANTIATED_BUILT_IN_MASK_0_31

Comment (robertsipka) [inline jerry-core/ecma/base/ecma-globals.h]: I've updated the patch based on your suggestion.

File: jerry-core/ecma/base/ecma-gc.c [hunk 6]
```diff
@@ -242,16 +242,6 @@ ecma_gc_mark_property (ecma_property_t *property_p) /**< property */
...
-        case ECMA_INTERNAL_PROPERTY_STRING_INDEXED_ARRAY_VALUES: /* a collection of ecma values */
-        {
-          JERRY_UNIMPLEMENTED ("Indexed array storage is not implemented yet.");
-        }
-
-        case ECMA_INTERNAL_PROPERTY_PROTOTYPE: /* the property's value is located in ecma_object_t
-                                                * (see above in the routine) */
-        case ECMA_INTERNAL_PROPERTY_EXTENSIBLE: /* the property's value is located in ecma_object_t
-                                                 * (see above in the routine) */
         case ECMA_INTERNAL_PROPERTY__COUNT: /* not a real internal property type,
```
Comment (akiss77) [inline jerry-core/ecma/base/ecma-gc.c]: This case has a very strange placement (to me, at least). Usually, such an "impossible" case comes last in a switch. Anyone else for or against moving this to the end?

Comment (zherczeg) [inline jerry-core/ecma/base/ecma-gc.c]: Lets move it.

Comment (robertsipka) [inline jerry-core/ecma/base/ecma-gc.c]: Updated.

**Comment by zherczeg** [comment]
LGTM

File: jerry-core/ecma/base/ecma-gc.c [hunk 7]
```diff
@@ -322,6 +305,12 @@ ecma_gc_mark_property (ecma_property_t *property_p) /**< property */
 
           break;
         }
+        case ECMA_INTERNAL_PROPERTY__COUNT: /* not a real internal property type,
+                                         * but number of the real internal property types */
```
Comment (LaszloLango) [inline jerry-core/ecma/base/ecma-gc.c]: wrong indentation

**Comment by LaszloLango** [comment]
LGTM after my comment is fixed. :)

**Comment by robertsipka** [comment]
Thanks, I've updated this patch.

**Pull request merged**